test/unit/index-lambda-cat-bot.test.js

```diff
@@ -1,4 +1,69 @@
-test('this is a placeholder test', () => {
-  console.log('hello, i am running this test!!');
-  expect(true).toBe(true);
+const { handler } = require('../../src/index-lambda-cat-bot');
+const index = require('../../src/index-lambda-cat-bot');
+const s3 = require('../../src/services/s3');
+const twitter = require('../../src/services/twitter');
+const fakeImages = require('../data/s3-response-list-objects-v2.json').Contents;
+const uploadImageAPIResponse = require('../data/twitter-response-upload-image.json');
+
+jest.mock('s3');
+jest.mock('twitter');
+
+afterEach(() => {
+  jest.resetAllMocks();
 });
+
+test('handler exists', () => {
+  expect(handler()).toBeDefined();
+})
+
+describe('Empty staging folder reaction', () => {
```

Comment:
Since you're testing the `handler` in each test here, let's reorganize how the `describe`s and `test`s are setup.

Personally, I like to follow this pattern where each `describe` encapsulates all tests for a single function. And each `test` under that `describe` tests a different path under that function:
```javascript
describe('myFunctionName1', () => {
  test('test happy path', async () => { ... });
  test('test unhappy path', async () => { ... });
})

describe('myFunctionName2', () => {
  test('test happy path', async () => { ... });
  test('test unhappy path', async () => { ... });
})
```

So in our case here, we can set it up this way:
```javascript
describe('handler', () => {
  test('Staging folder gets repopulated if it is empty', async () => { ... });
  test('twitter.uploadImage and twitter.createTweet are called if isTweetingEnabled is true', async () => { ... });
  test('Error from twitter calls handled', async () => { ... });
})
```

Reply:
Let me know what you think!

Reply:
Makes sense. That's much more organized than how I had it. I also followed your `happy path` then `unhappy path` by just switching the first and second tests